### PR TITLE
Sds 819 left nav

### PR DESCRIPTION
**Pull request by jaclynj**
https://meetup.atlassian.net/browse/SDS-835

image

### CODE REVIEW

File: packages/swarm-docs/gatsby-config.js [hunk 1]
```diff
@@ -4,13 +4,21 @@ module.exports = {
...
         path: `${__dirname}/src/images`,
       },
     },
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        name: 'pages',
+        path: `${__dirname}/src/pages/`,
+      },
+    },
```
Comment (jaclynj) [inline packages/swarm-docs/gatsby-config.js]: for querying files with graphql

File: packages/swarm-docs/gatsby-config.js [hunk 2]
```diff
@@ -4,13 +4,21 @@ module.exports = {
   },
   plugins: [
     'gatsby-plugin-react-helmet',
+    'gatsby-plugin-flow',
```
Comment (jaclynj) [inline packages/swarm-docs/gatsby-config.js]: so the nav file can use flow

File: packages/swarm-docs/src/components/layout.js [hunk 3]
```diff
@@ -37,23 +46,26 @@ const Layout = ({ children }) => {
...
-          <div
-            style={{
-              margin: '0 auto',
-              maxWidth: 960,
-              padding: '0px 1.0875rem 1.45rem',
-              paddingTop: 0,
-            }}
-          >
-            {children}
+          <Nav fileNames={ data.allFile.edges.map((value) => value.node.name)} />
```
Comment (jaclynj) [inline packages/swarm-docs/src/components/layout.js]: recommend changing your diff settings to more easily see the changes:
image

File: yarn.lock [hunk 4]
```diff
@@ -22,14 +21,12 @@
...
   version "7.0.0"
   resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.0.0.tgz#06e2ab19bdb535385559aabb5ba59729482800f8"
-  integrity sha512-OfC2uemaknXr87bdLUkWog7nYuliM9Ij5HUcajsVcMCpQrcLmtxRbVFTIqmcSkSeYRBFBRxs2FiUqFJDLdiebA==
   dependencies:
     "@babel/highlight" "^7.0.0"
 
 "@babel/core@^7.0.0", "@babel/core@^7.1.6":
   version "7.3.3"
   resolved "https://registry.yarnpkg.com/@babel/core/-/core-7.3.3.tgz#d090d157b7c5060d05a05acaebc048bd2b037947"
-  integrity sha512-w445QGI2qd0E0GlSnq6huRZWPMmQGCp5gd5ZWS4hagn0EiwzxD5QMFkpchyusAyVC1n27OKXzQ0/88aVU9n4xQ==
```
Comment (jaclynj) [inline yarn.lock]: Not sure what this is about - @hsbacot what version of yarn are you using?

File: packages/swarm-docs/src/components/nav.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import * as React from 'react';
+import { Link } from 'gatsby';
+
+type Props = {
+    fileNames: Array<string>
+};
+
+const Nav = (props: Props): React.Element<'div'> => {
+    const navItems = props.fileNames.map((name) => (
+        <div key={name}><Link to={`/${name}`} >{name}</Link></div>
```
Comment (hsbacot) [inline packages/swarm-docs/src/components/nav.js]: `text--link` should style the links with swarms styles

**Comment by hsbacot** [comment]
One thing to keep in mind for the future, we're using (conventional commits)[https://www.conventionalcommits.org/en/v1.0.0-beta.3/] for automated versioning and deploys.

**Pull request merged**